src/stream/src/executor/dispatch.rs

```diff
@@ -818,6 +862,126 @@ impl Dispatcher for BroadcastDispatcher {
     }
 }
 
+/// Dispatch stream chunk based on table name from upstream DB
+#[derive(Debug)]
+pub struct CdcTableNameDispatcher {
+    outputs: Vec<BoxedOutput>,
+    // column index to the `_rw_table_name` column
+    table_name_col_index: usize,
+    output_indices: Vec<usize>,
+    dispatcher_id: DispatcherId,
+    dispatcher_id_str: String,
+    downstream_table_name: Option<String>,
+}
+
+impl CdcTableNameDispatcher {
+    pub fn new(
+        outputs: Vec<BoxedOutput>,
+        table_name_col_index: usize,
+        output_indices: Vec<usize>,
+        dispatcher_id: DispatcherId,
+        downstream_table_name: Option<String>,
+    ) -> Self {
+        Self {
+            outputs,
+            table_name_col_index,
+            output_indices,
+            dispatcher_id,
+            dispatcher_id_str: dispatcher_id.to_string(),
+            downstream_table_name,
+        }
+    }
+}
+
+impl Dispatcher for CdcTableNameDispatcher {
+    async fn dispatch_data(&mut self, chunk: StreamChunk) -> StreamResult<()> {
+        let num_outputs = self.outputs.len();
```

Comment:
I believe it's only possible to have a single output here?

Reply:
I guess the case is similar to `SimpleDispatcher`.

https://github.com/risingwavelabs/risingwave/blob/e4b65a8c03216cabb8b99728c1c8cfc3b3632479/src/stream/src/executor/dispatch.rs#L988-L1000